src/porepy/utils/error.py

```diff
@@ -174,42 +180,68 @@ def fun_p(pt): return np.sin(2*np.pi*pt[0])*np.sin(2*np.pi*pt[1])
-):
-    """
-    Compute the L2 error of a scalar or vector field with respect to a reference
-    field. It is possible to compute the relative error (default) or the
-    absolute error.
+def l2_error(
+    grid: pp.GridLike,
+    true_array: np.ndarray,
+    approx_array: np.ndarray,
+    is_scalar: bool,
+    is_cc: bool,
+    relative: bool = False,
+) -> pp.number:
+    """Compute discrete L2-error as given in [1].
 
-    Parameters
-    ----------
-    g : grid
-        Grid, or a subclass, with geometry fields computed.
-    val : np.ndarray (dim of val, g.num_cells)
-        Scalar or vector field.
-    val_ex : np.ndarray (dim of val, g.num_cells)
-        Reference scalar or vector field, i.e. the exact solution
-    relative: bool (True default)
-        Compute the relative error (if True) or the absolute error (if False).
+    It is possible to compute the absolute error (default) or the relative error.
 
-    Return
-    ------
-    out: double
-        The L2 error of the input fields.
+    Parameters:
+        grid: Either a subdomain grid or a mortar grid.
+        true_array: Array containing the true values of a given variable.
+        approx_array: Array containing the approximate values of a given variable.
+        is_scalar: Whether the variable is a scalar quantity. Use ``False`` for
+            vector quantities. For example, ``is_scalar=True`` for pressure, whereas
+            ``is_scalar=False`` for displacement.
+        is_cc: Whether the variable is associated to cell centers. Use ``False``
+            for variables associated to face centers. For example, ``is_cc=True``
+            for pressures, whereas ``is_scalar=False`` for subdomain fluxes.
+        relative: Compute the relative error (if True) or the absolute error (if False).
 
-    Examples
-    --------
-    p = ...
+    Returns:
+        Discrete L2-error between the true and approximated arrays.
 
-    def fun_p(pt): return np.sin(2*np.pi*pt[0])*np.sin(2*np.pi*pt[1])
-    p_ex = interpolate(g, fun_p)
-    err_p = err_L2(g, p, p_ex)
+    Raises:
+        NotImplementedError if a mortar grid is given and ``is_cc=False``.
+        ZeroDivisionError if the denominator in the relative error is zero.
 
-    """
+    References:
 
-    val, val_ex = np.asarray(val), np.asarray(val_ex)
-    err = norm_L2(g, np.subtract(val, val_ex))
-    den = norm_L2(g, val_ex) if relative else 1
-    assert den != 0
-    return err / den
+        - [1] Nordbotten, J. M. (2016). Stable cell-centered finite volume
+          discretization for Biot equations. SIAM Journal on Numerical Analysis,
+          54(2), 942-968.
+
+    """
+    # Sanity check
+    if isinstance(grid, pp.MortarGrid) and not is_cc:
+        raise NotImplementedError("Interface variables can only be cell-centered.")
+
+    # Obtain proper measure, e.g., cell volumes for cell-centered quantities and face
+    # areas for face-centered quantities.
+    if is_cc:
```

Comment:
```suggestion
    if is_cc:
            meas = grid.cell_volumes
    else:
        meas = grid.face_areas
    if not is_scalar:
        meas = meas.repeat(grid.dim)

```
I think this will work, and it is more concise.

Reply:
Do check the indentations, though.

Reply:
Beautiful. I have incorporated the suggestion (but not directly due to indentation issues). Thanks!